iothub/device/src/WritableClientProperty.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Azure.Devices.Shared;
+
+namespace Microsoft.Azure.Devices.Client
+{
+    /// <summary>
+    /// The writable property update request received from service.
+    /// </summary>
+    /// <remarks>
+    /// A writable property update request should be acknowledged by the device or module by sending a reported property.
+    /// This type contains a convenience method to format the reported property as per IoT Plug and Play convention.
+    /// For more details see <see href="https://docs.microsoft.com/azure/iot-develop/concepts-convention#writable-properties"/>.
+    /// </remarks>
+    public class WritableClientProperty
+    {
+        internal WritableClientProperty()
+        {
+        }
+
+        /// <summary>
+        /// The value of the writable property update request.
+        /// </summary>
+        public object Value { get; internal set; }
+
+        /// <summary>
+        /// The version number associated with the writable property update request.
+        /// </summary>
+        public long Version { get; internal set; }
+
+        internal PayloadConvention Convention { get; set; }
+
+        /// <summary>
+        /// Creates a writable property update response that can be reported back to the service.
+        /// </summary>
+        /// <remarks>
+        /// This writable property update response will contain the property value and version supplied in the writable property update request.
+        /// If you would like to construct your own writable property update response with custom value and version number, you can
+        /// create an instance of <see cref="PayloadSerializer.CreateWritablePropertyResponse(object, int, long, string)"/>.
+        /// See <see href="https://docs.microsoft.com/azure/iot-develop/concepts-convention#writable-properties"/> for more details.
+        /// </remarks>
+        /// <param name="statusCode">An acknowledgment code that uses an HTTP status code.</param>
+        /// <param name="description">An optional acknowledgment description.</param>
+        /// <returns>A writable property update response that can be reported back to the service.</returns>
+        public IWritablePropertyResponse AcknowledgeWith(int statusCode, string description = default)
+        {
+            return Convention.PayloadSerializer.CreateWritablePropertyResponse(Value, statusCode, Version, description);
```

Comment:
>Value [](http://example.com/codeflow?start=79&length=5)

This assumes that you're acknowledging the exact value that was sent by the service.  What if you want to respond with a value that is slightly off?  For exmple, the service sends a writable property request to set the temperature to 72.3 degrees, and the device app returns a WritablePropertyResponse that says "we set it to 72.2 degrees" because the hardware had to round down.  (I assume this is a valid scenario, but I'm assuming and you know what that means :))

Reply:
That's fair. We've included the following doc comment in the ack function which calls out how to achieve this:
https://github.com/Azure/azure-iot-sdk-csharp/blob/7cc63d2ac338c3527d21329d5fe5add50e80653b/iothub/device/src/WritableClientProperty.cs#L38-L41